### PR TITLE
add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by mohamedelshamy55**
In this project, you will start building your school library app. In this initial step, you will implement the classes to represent students and teachers.
Implement classes and objects in Ruby.
Implement encapsulation and inheritance with Ruby.
Run a program using the command line.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Person
+  def initialize(_id, age, name = 'unknown', parent_permission: true)
```
Comment (er-contreras) [inline person.rb]: - [ ] it's not necessary to include "id" as a parameter. The id is part of the attributes and it's generated automatically.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Student < Person
+  def initialize(_clasroom, age, name = 'unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```
Comment (er-contreras) [inline student.rb]: - [ ] There is a typo in "_clasroom" should be "classroom" to match with the instance variable. You don't need to add the underscore sign in this case.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def can_use_service
+    of_age || parent_permission == true
+  end
```
Comment (er-contreras) [inline person.rb]: - [ ] It is a code style convention Method "can_use_service" should include "?" at the end as follow "can_use_service?" that way we are telling that method to return just true or false.
- [ ] "parent_permission" should be called as an instance variable as follow @parent_permission without "== true" if not the variable does not be called.

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+  private
+
+  def of_age
+    @age >= 18
+  end
```
Comment (er-contreras) [inline person.rb]: - [ ] This method should be called as It is a code style convention to boolean values

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Teacher < Person
+  def initialize(_specialization, age, name = 'unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @specialization = specialization
+  end
+
+  def can_use_service
+    true
+  end
```
Comment (er-contreras) [inline teacher.rb]: - [ ] After fixing the last reviews check in this class that you are able to create a teacher object to be sure that everything is working well. 😉

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @mohamedelshamy55, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

almost there

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @mohamedelshamy55,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mohamedelshamy55** [comment]
@TedLivist and @eri8-9 Thank you for finding time to review my project. I appreciate.

**Pull request merged**